src/App.tsx

```diff
@@ -14,11 +14,125 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  isReversed: boolean;
+  sortBy: string | null;
+  isRun: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isReversed: false,
+    sortBy: null,
+    isRun: false,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  start = () => {
+    this.setState((state) => ({
+      isRun: !state.isRun,
+    }));
+  };
+
+  sortByABC = () => {
+    this.setState({
+      sortBy: 'abc',
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      goods: goodsFromServer,
+      isReversed: false,
+      sortBy: null,
+    });
+  };
+
+  render() {
+    const {
+      goods,
+      isReversed,
+      sortBy,
+      isRun,
+    } = this.state;
+    const goodsForApp = [...goods];
+
+    if (sortBy) {
+      goodsForApp.sort((a, b) => {
+        switch (sortBy) {
+          case 'abc':
+            return a.localeCompare(b);
+          case 'length':
+            return a.length - b.length;
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (isReversed) {
+      goodsForApp.reverse();
+    }
+
+    return (
+      <div className="app">
+        <h1 className="app__title">Goods</h1>
+
+        <button
+          type="button"
+          className="app__start-reset"
+          onClick={() => {
+            if (!isRun) {
+              this.start();
+            } else {
+              this.reset();
+            }
+          }}
+        >
+          {!isRun ? 'Start' : 'Reset'}
+        </button>
```

Comment:
I think it's better to create separate buttons for each case than use condition inside onClick